lightning/src/routing/router.rs

```diff
@@ -919,16 +916,12 @@ where L::Target: Logger {
 					// Taking too many smaller paths also increases the chance of payment failure.
 					// Thus to avoid this effect, we require from our collected links to provide
 					// at least a minimal contribution to the recommended value yet-to-be-fulfilled.
-					//
-					// This requirement is currently 5% of the remaining-to-be-collected value.
-					// This means as we successfully advance in our collection,
-					// the absolute liquidity contribution is lowered,
-					// thus increasing the number of potential channels to be selected.
+					// This requirement is currently 10% of the value.
 
-					// Derive the minimal liquidity contribution with a ratio of 20 (5%, rounded up)
+					// Derive the minimal liquidity contribution with a ratio of 10 (10%, rounded up)
 					// or 100% if we're not allowed to do multipath payments.
 					let minimal_value_contribution_msat: u64 = if allow_mpp {
-						(recommended_value_msat - already_collected_value_msat + 19) / 20
+						(final_value_msat + 9) / 10
```

Comment:
First (trivial) approach is to fix the `minimal_value_contribution` to be a static 10% of `final_value_msat`. This however means that we may end up needlessly discarding candidate paths, even when there would still be 'room' towards the 10-path limit. Will investigate further if there is a good way to dynamically get more permissive.